clients/drcachesim/tracer/instr_counter.cpp

```diff
@@ -550,9 +550,9 @@ compute_irregular_trace_windows(std::vector<instr_interval_t> &instr_intervals)
     // must have a duration long enough to cover the end of the program.
     irregular_window_ptr =
         (irregular_window_t *)dr_global_alloc(sizeof(irregular_window_t));
-    // DELAY_FOREVER_THRESHOLD might be too small for long traces, but it doesn't matter
-    // because we trace_for_instrs = 0, so no window is created anyway.
-    irregular_window_ptr->no_trace_for_instrs = DELAY_FOREVER_THRESHOLD;
+    // This results in has_instr_count_threshold_to_enable_tracing() returning 0, which
```

Comment:
I don't fully understand what the return value of `has_instr_count_threshold_to_enable_tracing()` means. If it return false, does it mean "there doesn't existing any instr count threshold to enable tracing, therefore the caller should feel free to add tracing instrumentation", or does it mean "there exists an instr count threshold to enable tracing, but it hasn't reached yet"?

Could you document that in a comment?

Reply:
My interpretation is the latter: "there exists (if true is returned) an instr count threshold to enable tracing".
But this function doesn't tell you anything about "reaching the instr count".
So, when it returns false, it means that "there is no instr count to enable tracing".
Adding a comment.